server/plugins/client/bundle.js

```diff
@@ -14,20 +28,27 @@ var parallel = require('async').parallel
  * and avoiding unneeded bundling with browserify saves a significant time.
  */
 function bundleClient (hoodieClientPath, bundleTargetPath, config, callback) {
-  parallel([
+  var plugins = [
+    path.resolve(config.paths && config.paths.applicationRoot || '', 'hoodie/client')
+  ].filter(checkModule)
+  var getPluginsModifiedTimes = plugins.map(function (pluginPath) {
+    return getModifiedTime.bind(null, requireResolve(pluginPath))
+  })
```

Comment:
> `getPluginsModifiedTimes` sounds like a method, but it’s an array with values, right? Can we rename it to `pluginsModifiedTimes`?

No, it's an array (of one at the moment, but with the intention that it is ready to expand to accept custom plugins) of bound instances of the `getModifiedTime` function, which are then run in parallel with the other bound instances. I did debate over the naming of that one - how do you name an array of functions? Maybe taking a cue from FTP, how about `mgetPluginsModifiedTimes`?

> If I understand it correctly, we get the entry JavaScript file for the app-specific client plugin, usually that would be `hoodie/client/index.js` or `hoodie/client.js`. 

> The problem that we have is that while the client bundle will always be one file, the app-specific plugin (and 3rd party plugins in future) will most be spread around multiple files. I don’t have an idea how take this into account right now, I think this might end up being a very hard problem to solve :/ Maybe it’s something that browserify itself can do, with a custom plugin or something?

I think hooking into the browserify API may indeed be the way to go - browserify uses [module-deps](https://github.com/substack/module-deps) internally to calculate the dependency tree, and allows emitting the output from this as an event on it's internal [pipeline](https://github.com/substack/node-browserify#bpipeline). If we save the output from this as a JSON file alongside the bundle, we can load this in, recheck all of the dependencies for modifications, and regenerate the dependency tree each time the server is started, with little extra cost due to it being work browserify is doing anyway. Sound like a plan?